### DISCUSSION TITLE
Limites pour branching "Condition si"

### DISCUSSION THREAD

**Discussion by jodionrq** [MTESSDev/FRW#25] [Annonces]
Bonjour,

Est-ce qu'il y a une limite au branching lorsqu'on utilise des "Condition si". Par exemple, dans la situation du P700U (image jointe), lorsqu'on fait apparaitre un groupe dans lequel il y a des boutons radio, est-ce ces boutons radio peuvent à leur tour faire apparaitre du texte? Dans l'exemple illustré, ça voudrait dire de faire apparaitre du texte en choisissant "Oui" ou "Non".

Merci!
![Condition_si](https://user-images.githubusercontent.com/124822406/225056316-ffd6868e-3709-4dcd-9709-60142c1d0843.png)

**Comment by j0ber** [comment]
Bonjour,

Oui c'est possible voir cette exemple à la page https://formulaires.it.mtess.gouv.qc.ca/Form/1/P700u/0/N/#p=23

Une condition v-if peut s'appliquer sur n'importe quel contrôle. Dans le cas d'un texte, normalement il va être dans un contrôle de type "dynamic".  À noter que dans l'exemple le contrôle bouton radio, et le texte sont les deux à l'intérieur d'un contrôle "dynamic" mais ce n'est pas nécessaire.

![image](https://user-images.githubusercontent.com/26974817/225073701-c2d93ab7-d18e-4263-b2e8-015e0744bb14.png)

**Reply by jodionrq** [reply]
Merci pour la réponse rapide.
Ça fonctionne en présence d'un contrôle "type: dynamic" (1er exemple), mais lorsqu'on les intègrent à l'intérieur un contrôle "type: group" (2e exemple), les conditions v-if ne s'appliquent pas.
Est-ce que c'est le contrôle "group" qui empêche le fonctionnement? Est-ce qu'il faut intégrer un contrôle "dynamic" dans le code?

![type_dynamic](https://user-images.githubusercontent.com/124822406/225100156-2def3d40-d550-45a8-8164-f5d9a8df4f05.png)

![type_group](https://user-images.githubusercontent.com/124822406/225100222-197787e9-6c78-4e1f-ae4c-8f902cb59431.png)

**Comment by j0ber** [comment]
En fait c'est le v-if qui n'est pas correct dans cette situation. Lorsqu'on fait référence à un groupe la méthode val doit être utilisée comme suit:

`val('nomGroupe.nomChamp', indexeOccuranceGroupeSiApplicable)` 

Dans votre cas le groupe n'est pas répétable donc :  
`val('nomGroupe.nomChamp')` 

Je constate que nous n'avons pas d'exemple de ce type. Nous allons en ajouter un.

**Reply by jodionrq** [reply]
Super ça fonctionne! Merci beaucoup c'est très apprécié.

**Reply by j0ber** [reply]
S.V.P. indiquer la réponse comme étant répondue :)   "Mark as answer"
